metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
I believe that we should use polymorphism to provide `parse(byte[], SensorParserConfig)` with a default implementation for `parse(byte[])`.  I think that should be done in the `MessageParser` interface using the `default` keyword.  We have done precisely this in that interface for `parseOptional`.  This would provide people the ability to override either one without negatively affecting existing parsers.

As for calling configure per message, this would change the semantics of the configure method for existing parsers (i.e. shifting from configure being a one-time thing to a per-message thing might change how people use the method regarding possibly computationally intensive setup).  

As a side-note, I think we should dump the `init()` method from that interface and just adopt `configure`.  It's confusing having two setup methods with precisely the same semantic characteristics.